src/averageAges.js

```diff
@@ -15,11 +15,15 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = century
+    ? people
+      .filter(person => person.sex === 'm'
+       && Math.ceil(person.died / 100) === century)
+    : people.filter(person => person.sex === 'm');
+
+  const menAge = men.map(person => person.died - person.born);
+
+  return menAge.reduce((sum, age) => sum + age) / menAge.length;
```

Comment:
Map is redundant here, you can calculate (person.died - person.born) in reduce method